src/debugAdapter/goDebug.ts

```diff
@@ -781,6 +776,11 @@ class GoDebugSession extends LoggingDebugSession {
 
 	protected disconnectRequest(response: DebugProtocol.DisconnectResponse, args: DebugProtocol.DisconnectArguments): void {
 		log('DisconnectRequest');
+		// For remote process, we have to issue a continue request
+		// before disconnecting.
+		if (this.delve.request === 'attach' && !this.delve.debugProcess) {
```

Comment:
request can be launch and the user could be using the old of way doing remove debugging.

I would suggest adding a property `isLocalDebugging` on `this.delve` so that we can use that directly instead of checks like this